### PR TITLE
Add option for primary key constraint detection

### PR DESCRIPTION
**Pull request by nnsgmsone**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [X] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #

## What this PR does / why we need it:

### CODE REVIEW

**Review by XuPeng-SH — APPROVED** [review]
lgtm

**Review by fengttt — COMMENTED** [review]
Do you need to handle delete as well?

What if you are deleting a non existing key?   Is that considered OK?

**Comment by nnsgmsone** [comment]
> Do you need to handle delete as well?
> 
> What if you are deleting a non existing key? Is that considered OK?

yes, then when I read it, I will go find the rowid corresponding to the fast on s3 and remove the row.

File: pkg/vm/engine/disttae/logtail.go [hunk 1]
```diff
@@ -80,15 +82,19 @@ func consumerCheckPoint(ckpt string) error {
 	return nil
 }
 
-func consumerEntry(idx, primaryIdx int, ctx context.Context, db *DB, mvcc MVCC, e *api.Entry) error {
+func consumerEntry(idx, primaryIdx int, tbl *table, ts timestamp.Timestamp,
+	ctx context.Context, db *DB, mvcc MVCC, e *api.Entry) error {
 	if e.EntryType == api.Entry_Insert {
 		if isMetaTable(e.TableName) {
-			return db.getMetaPartitions(e.TableName)[idx].Insert(ctx, -1, e.Bat)
+			if err := tbl.parts[idx].DeleteByBlockID(ctx, ts, e.BlockId); err != nil {
```
[the remark this answers is not in the document]
Comment (fengttt) [reply pkg/vm/engine/disttae/logtail.go:89]: The name is really, really weird.   But OK.

**Review by fengttt — APPROVED** [review]

**Pull request merged**